source/hic_hal/freescale/k20dx/gpio.c

```diff
@@ -57,24 +57,36 @@ void gpio_init(void)
     PIN_nRESET_GPIO->PDDR &= ~PIN_nRESET;
     PIN_nRESET_PORT->PCR[PIN_nRESET_BIT] = PORT_PCR_MUX(1);
 
-    // Keep powered off in bootloader mode
-    // to prevent the target from effecting the state
-    // of the reset line / reset button
     if (!daplink_is_bootloader()) {
         // configure pin as GPIO
         PIN_POWER_EN_PORT->PCR[PIN_POWER_EN_BIT] = PORT_PCR_MUX(1);
-        // force always on logic 1
-        PIN_POWER_EN_GPIO->PDOR |= 1UL << PIN_POWER_EN_BIT;
-        PIN_POWER_EN_GPIO->PDDR |= 1UL << PIN_POWER_EN_BIT;
+        // set output to 0
+        PIN_POWER_EN_GPIO->PCOR = PIN_POWER_EN;
+        // switch gpio to output
+        PIN_POWER_EN_GPIO->PDDR |= PIN_POWER_EN;
     }
+}
 
-    // Let the voltage rails stabilize.  This is especailly important
-    // during software resets, since the target's 3.3v rail can take
-    // 20-50ms to drain.  During this time the target could be driving
-    // the reset pin low, causing the bootloader to think the reset
-    // button is pressed.
-    // Note: With optimization set to -O2 the value 1000000 delays for ~85ms
-    busy_wait(1000000);
+void gpio_set_board_power(bool powerEnabled)
+{
+    if (!daplink_is_bootloader()) {
```

Comment:
Why is this turned off for bootloaders? Seems like it applies equally to bootloader an interface, just the interface code is the one which will turn it on.

Reply:
Partially a carry-over from the original code. Also, "board power" may mean different things to different boards. On some boards, I can imagine you would need the bootloader to turn on some power. So the bootloader might call this function just like the interface, and let the board decide whether to turn on power in the bootloader.

But I'll remove the bootloader check. Right now the bootloader doesn't call this function, and there are no boards that require additional power for the bootloader.